### PR TITLE
Test to verify correct functionality

### PR DESCRIPTION
**Pull request by martin-morek**
- Fixes [570](https://github.com/ably/ably-java/issues/570)
- Added test should reproduces bug described in the issue, however it doesn't it only proves everything works correctly

### CODE REVIEW

File: lib/src/test/java/io/ably/lib/test/realtime/RealtimeChannelTest.java [hunk 1]
```diff
@@ -3,6 +3,7 @@
 import io.ably.lib.debug.DebugOptions;
 import io.ably.lib.realtime.*;
```
Comment (paddybyers) [inline lib/src/test/java/io/ably/lib/test/realtime/RealtimeChannelTest.java:4]: I realise this code existed before this PR, but these are inconsistent import styles. I think generally we're avoiding wildcard import statements.

File: lib/src/test/java/io/ably/lib/test/realtime/RealtimeChannelTest.java [hunk 2]
```diff
@@ -196,6 +191,58 @@ public void attach_with_channel_params_set_options() {
...
+
+            final Map<String, String> params = new HashMap<>();
+            params.put("rewind", "1");
+            final ChannelOptions options = new ChannelOptions();
+            options.params = params;
+
+            Channel channel = ably.channels.get(channelName, options);
+            ablyRest.channels.get(channelName).publish("", "test-message");
+
+            // channel is in initialized state when created
```
Comment (paddybyers) [inline lib/src/test/java/io/ably/lib/test/realtime/RealtimeChannelTest.java:217]: Inconsistent comment style

File: lib/src/test/java/io/ably/lib/test/realtime/RealtimeChannelTest.java [hunk 3]
```diff
@@ -196,6 +191,58 @@ public void attach_with_channel_params_set_options() {
...
+
+            messageWaiter.waitFor(1);
+            // when message is received channel state is already in attached state
+            assertEquals(ChannelState.attached, channel.state);
+
+            ably.channels.release(channelName);
+
+            // verify there are no channels after realising
+            assertFalse(ably.channels.entrySet().iterator().hasNext());
+            Thread.sleep(5000);
```
Comment (paddybyers) [inline lib/src/test/java/io/ably/lib/test/realtime/RealtimeChannelTest.java:232]: Why is this `sleep()` needed?

Edit: I've read the issue discussion now that this was based on. Have you managed to get this test to fail before the fix?

File: lib/src/test/java/io/ably/lib/test/realtime/RealtimeChannelTest.java [hunk 4]
```diff
@@ -196,6 +191,58 @@ public void attach_with_channel_params_set_options() {
...
+
+            // get channel again
+            channel = ably.channels.get(channelName, options);
+            assertEquals(ChannelState.initialized, channel.state);
+        } catch (AblyException | InterruptedException e) {
+            e.printStackTrace();
+            fail("init0: Unexpected exception instantiating library");
```
Comment (paddybyers) [inline lib/src/test/java/io/ably/lib/test/realtime/RealtimeChannelTest.java:239]: Again, I realise this is a cut+paste of the same error elsewhere, but this message should refer to the correct test name, or not refer to the name at all

File: lib/src/test/java/io/ably/lib/test/realtime/RealtimeChannelTest.java [hunk 5]
```diff
@@ -196,6 +191,58 @@ public void attach_with_channel_params_set_options() {
...
+            // channel is not attached yet
+            assertEquals(ChannelState.attaching, channel.state);
+
+            messageWaiter.waitFor(1);
+            // when message is received channel state is already in attached state
+            assertEquals(ChannelState.attached, channel.state);
+
+            ably.channels.release(channelName);
+
+            // verify there are no channels after realising
```
Comment (paddybyers) [inline lib/src/test/java/io/ably/lib/test/realtime/RealtimeChannelTest.java:230]: typo `realising` -> `releasing`

**Review by paddybyers — COMMENTED** [review]
This PR is still draft so this isn't a final review but I added some comments.

Please also make commit messages more informative - at least reference the issue ID or state what functionality is being fixed.

File: lib/src/test/java/io/ably/lib/test/realtime/RealtimeChannelTest.java (hunk 3 above)
Comment (martin-morek) [reply lib/src/test/java/io/ably/lib/test/realtime/RealtimeChannelTest.java:232]: No, added sleeping didn't changed anything but I left it here for demonstration.

File: lib/src/test/java/io/ably/lib/test/realtime/RealtimeChannelTest.java (hunk 1 above)
Comment (martin-morek) [reply lib/src/test/java/io/ably/lib/test/realtime/RealtimeChannelTest.java]: Thanks for pointing it out, fixed.

File: lib/src/test/java/io/ably/lib/test/realtime/RealtimeChannelTest.java (hunk 4 above)
Comment (martin-morek) [reply lib/src/test/java/io/ably/lib/test/realtime/RealtimeChannelTest.java:252]: Fail message reflects on scenario when Ably cannot be constructed. Or are you referring to something else?

Comment (paddybyers) [reply lib/src/test/java/io/ably/lib/test/realtime/RealtimeChannelTest.java:252]: My comment was just that `init0` was the name of a test. This line of code got copied into multiple other tests, without updating the test name. Either update it, or remove it.

File: lib/src/test/java/io/ably/lib/test/realtime/RealtimeChannelTest.java (hunk 1 above)
Comment (QuintinWillison) [reply lib/src/test/java/io/ably/lib/test/realtime/RealtimeChannelTest.java]: I thought we had this covered by checkstyle... but perhaps that doesn't run over the test classes. 🤔

**Review by QuintinWillison — CHANGES_REQUESTED** [review]
I'm really not sure what value this pull request adds to the codebase, so I'm inclined to simply close it. Happy to be persuaded otherwise so suggestions appreciated.

**Comment by QuintinWillison** [comment]
I notice no objections to my suggestion of closing this PR, so I'm doing that now.

**Pull request closed without merging**